internal/wasm/store.go

```diff
@@ -196,11 +209,24 @@ type (
 		Buffer []byte
 		Min    uint32
 		Max    *uint32
+
+		// address is the memoryaddr(https://www.w3.org/TR/wasm-core-1/#syntax-tableaddr) of this memory instance.
+		// In other words, this equals the index of this memory instance in store.Memories.
```

Comment:
probably better to call this `index` and use the existing type `wasm.Index` vs the confusing uint64 which is at best an implementation detail of additions.

```suggestion
		// address is the index of this memory instance in Store.Memories.
		// See See https://www.w3.org/TR/2019/REC-wasm-core-1-20191205/#syntax-tableaddr
```